encoding/thrift/thriftrw-plugin-yarpc/server.go

```diff
@@ -130,6 +130,20 @@ func (h handler) <.Name>(ctx <$context>.Context, body <$wire>.Value) error {
 	return h.impl.<.Name>(ctx, <range .Arguments>args.<.Name>,<end>)
 }
 <else>
+
+func (h handler) <.Name>_applicationErrorName(err error) string {<if .Exceptions>
+    switch err.(type) {
+		<range .Exceptions ->
+		case <formatType .Type>:
+			if v, ok := err.(<formatType .Type>); ok {
+				return v.ErrorName()
+			}
+		<end ->
+	}
+<end>
+	return "unknown-error"
```

Comment:
I think it's preferable to leave this as an empty string.

Reply:
There would be cases in which the destination service might return a generic Go error, which is not listed as part of the Thrift contract.
I don't know exactly how that would translate to the middleware, but I am assuming it wouldn't be able to match the error with any of the ones listed in Thrift. In my opinion, it would be better to return "unknown-error" for such cases, but I don't mind returning an empty string instead, just telling you my reasoning for this. Do you think it's preferable to return an empty string even for these cases?

Reply:
I thought about a little more. Having the metric as "unknown-error" does not really bring any value, therefore I am always returning an empty string. The observability middleware will only really include this information if it's different than an empty string.